cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1798,23 +1798,22 @@ pub(crate) fn define(
     }
 
     // SIMD extractlane
-    let mut x86_pextr_mapping: HashMap<u64, (&'static [u8], Option<SettingPredicateNumber>)> =
-        HashMap::new();
-    x86_pextr_mapping.insert(8, (&PEXTRB, Some(use_sse41_simd)));
-    x86_pextr_mapping.insert(16, (&PEXTRW_SSE2, None));
-    x86_pextr_mapping.insert(32, (&PEXTR, Some(use_sse41_simd)));
-    x86_pextr_mapping.insert(64, (&PEXTR, Some(use_sse41_simd)));
+    let mut x86_pextr_mapping: HashMap<u64, &'static [u8]> = HashMap::new();
+    x86_pextr_mapping.insert(8, &PEXTRB);
+    x86_pextr_mapping.insert(16, &PEXTRW);
+    x86_pextr_mapping.insert(32, &PEXTR);
+    x86_pextr_mapping.insert(64, &PEXTR);
 
     for ty in ValueType::all_lane_types().filter(allowed_simd_type) {
-        if let Some((opcode, isap)) = x86_pextr_mapping.get(&ty.lane_bits()) {
+        if let Some(opcode) = x86_pextr_mapping.get(&ty.lane_bits()) {
```

Comment:
(unrelated, feel free to dismiss and we can do it in a follow-up, but) I realized we can avoid the HashMap with code doing:

```
let encoding = match ty.lane_bits() {
  8 => &PEXTRB,
  16 => &PEXTRW,
  ...
  _ => { continue; } // or a panic if that makes sense, but i don't think so
};
```

Reply:
Sounds good to me: https://github.com/CraneStation/cranelift/issues/1091.